File: it2java/oop/Point.java
public class Point {
    // instance variables
    int x, y;

    // Constructor:
    Point(int x, int y) {
        // local to constructor
        this.x = x;
        this.y = y;
        System.out.println("Point Created");
    }

    void show() {
        // ( 5, 7 )
        System.out.println("( " + x + " , " + y + " )");

    }

    void calcDistance(Point p2) {
        // p1 and p2
        // this -> current invoking
        double d = Math.pow((p2.x - this.x), 2) + Math.pow(p2.y - this.y, 2);

        double answer = Math.sqrt(d);
        System.out.println(
                "Distance between ( " + this.x + "," + this.y + " ) and ( " + p2.x + " ," + p2.y + " ) is: " + answer);

    }

    Point calMidpoint(Point p2) {
        // this and p2

        int midX = (this.x + p2.x) / 2;
        int midY = (this.y + p2.y) / 2;
        Point midPoint = new Point(midX, midY);
        return midPoint;

    }

}

File: it2python/oop/point.py
import math
class Point:
    def __init__(self,x,y):
        self.x=x
        self.y=y 
        print("Point Created: ")

    def show(self):
        print("( ",self.x,",",self.y," )")
    
    def calc_distance(self,point2):
        powers=math.pow((self.x-point2.x),2)+math.pow((self.y-point2.y),2)

        result=math.sqrt(powers)
        return result

    def print_mid_point(self,point2):
        midx=(self.x+point2.x)/2
        midy=(self.y+point2.y)/2
        midPoint=Point(midx,midy)
        return midPoint
        




#point 1
        
point1= Point(4,6)
point1.show()
print(type(point1))

point2=Point(89,32)

point2.show()

dis=point1.calc_distance(point2)
print("distance ",dis)

point3=Point(4,6)
point4=Point(6,7)
result=point3.calc_distance(point4)
point3.show()
point4.show()
print("distance ",result)

point3.print_mid_point(point4)

p5=Point(10,10)
p6=Point(1,1)

midPoint=p5.print_mid_point(p6)
midPoint.show()

result1=midPoint.calc_distance(p5)
print(result1)